model/library/objects.py

```diff
@@ -0,0 +1,313 @@
+
+    def stop(self):
+        """Stop the Instrumnet"""
+        self.is_running = InstrumentRunState.STOPPED
+
+class Stream:
+    """We have a stream (aka. jet) which shifts around in accord with these params.
+    The stream_shift_time_slice is a bit obscure. The idea is that.
+
+    Warning: The stream shift amount should be an integer multiple of the beam_shift_amount,
+    otherwisethe likelihood that they overlap (based on acuity) will be reduced.
+    Usually these will be the same."""
+
+    stream_shift_amount = 0.01  # Minimal unit of stream shift
+    p_stream_shift = 0.15  # prob. of stream shift per cycle
+    # A crazy ivan is when the stream goes haywire; It should happen very rarely.
+    p_crazy_ivan = 0.0001  # About 0.0001 gives you one/10k
+    crazy_ivan_shift_amount = round(random.uniform(0.1, 0.2), 2)  # 0.2
+    # If the beam doesn't hit a wall before this, we cut the run off here.
+    default_max_cycles = 10000
+    stream_pos = 0.0
+    allow_response_cycle = 99999999999
+    cycle = 1
+
+    def __init__(self):
+        pass
+
+class Beam:
+    """And the beam, which is under the control of the operator (or automation),
+    which can be shifted in accord with these params:"""
+     # You may want to have more or less fine control of the beam vs. the stream's shiftiness
+    beam_shift_amount = 0.01
+
+    # There are two different and wholly separate senses of acuity:
+    # physical_acuity: Whether the beam is physically on target, and functional_acuity:
+    # whether the operator can SEE that it is!
+    # Nb. Whole scale is -1...+1
+    # You want this a little larger than the shift so that it allows for near misses
+    physical_acuity = 0.02
+    beam_pos = 0.0
+
+class DataPoint:
+    """Object to store all the data for each beam point"""
+    quality:float = None
+
+    def __init__(self, quality):
+        self.quality = quality
+
+    def __str__(self):
+        return f'{self.quality:.2f}'
+
+class SampleData:
+    """Store all information about the sample as well as the datapoints
+    Sample attributes are stored in dict sampleTypeMapper
+    Preformance Quality, different natural signal response
+    when data is on peak data * PQ"""
+    data:List[DataPoint] = []
+
+    # TODO: make weights affect rescheduling
+    # TODO: What was this N shaped dynamics
+    def __init__(self, preformance_quality:float,
+            weight:float, setup_time:timedelta):
+        self.compleated:bool = False
+        self.timeout:bool = False
+        self.preformance_quality:float = preformance_quality
+        self.weight:float = weight
+        self.setup_time:timedelta = setup_time
+        self.data:List[DataPoint] = []
+        self.count:float = 0.0
+        self.mean:float = 0.0
+        self.m_2:float = 0.0
+        self.variance:float = 0.0
+        self.sdev:float = 0.0
+        self.err:float = 0.0
+        self.count_array = []
+        self.err_array = []
+
+    def append(self, data:DataPoint):
```

Comment:
# Important! 
This is the main place that I think there is a problem, the append function is called by the instrument to add a new datapoint to the sample, this also does all the mean and error calculations one of the three places were the bug might be